sources/views/table.js

```diff
@@ -0,0 +1,40 @@
+import {JetView} from "webix-jet";
+
+export default class Table extends JetView {
+	constructor(app, name, data) {
+		super(app, name);
+		this._gridData = data;
+	}
+
+	config() {
+		return {
+			rows: [
+				{
+					view: "datatable",
+					autoConfig: true,
+					localId: "table"
+				},
+				{
+					cols: [
+						{view: "button", value: "Add", click: () => this.addItem()},
+						{view: "button", value: "Delete", click: () => this.deletedItem()}
+					]
+				}
+			]
+		};
+	}
+
+	init(view) {
+		view.queryView("datatable").parse(this._gridData);
+	}
+
+	addItem() {
+		this.$$("table").add({Name: "name"});
+	}
+
+	deletedItem() {
+		const table = this.$$("table");
```

Comment:
You can move initialization variables like this in 'init' function. And you can use variable instead of using webix id in other funtions.
So, you can type something like following code in 'init' function
`this.tableComponent = this.$$("table")`
And after that you can type just
`const id = this.tableComponent.getSelectedId()`

This soultion can help you avoid mistakes with incorrect ids in future